lib/views/signin_page/alert_page.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:devfest/utils/colors.dart';
+import 'package:devfest/utils/extensions/extensions.dart';
+import 'package:flutter/material.dart';
+import 'package:gap/gap.dart';
+
+import '../../widgets/app_bar.dart';
+import '../../widgets/button.dart';
+
+enum AlertType { success, error, failure, almost }
+
+class AlertParams {
+  final AlertType type;
+  final String title;
+  final String description;
+  final void Function()? primaryAction;
+  final void Function()? secondaryAction;
+  final String? primaryBtnText;
+  final String? secondaryBtnText;
+
+  AlertParams({
+    required this.type,
+    required this.title,
+    required this.description,
+    this.primaryAction,
+    this.secondaryAction,
+    this.primaryBtnText,
+    this.secondaryBtnText,
+  })  : assert((primaryAction != null && primaryBtnText != null) ||
+            (primaryAction == null && primaryBtnText == null)),
```

Comment:
I think there are better ways to do this without using assert and loosing flexibilty

Reply:
Attended to